File: hw/bonilla/U3/HW19AbstractFactory/GUIFactoryJs/controller/GUIFactory.js
class GUIFactory {
    static getFactory() {
        // 🧠 Mover los require aquí evita la dependencia circular
        const WinFactory = require('../model/WinFactory');
        const LinuxFactory = require('../model/LinuxFactory');


        const sys = GUIFactory.readFromConfigFile("OS_TYPE");
        if (sys === 0) {
            return new WinFactory();
        } else {
            return new LinuxFactory();
        }
    }

    static readFromConfigFile(osType) {
        console.log("ReadFile");
        return 0; // 0 = Windows, 1 = Linux
    }

    createButton() {
        throw new Error("createButton() must be implemented");
    }

    createMenu() {
        throw new Error("createMenu() must be implemented");
    }
}

module.exports = GUIFactory;

File: hw/suarez/U2/design pattern/HW18AbstractFactory/src/ec/edu/espe/model/GUIFactory.java
package ec.edu.espe.model;

/**
 * 
 * @author Tammy Caizapanta
 */

public abstract class GUIFactory {
    
    public static GUIFactory getFactory() {
        int sys = readFromConfigFile("OS_TYPE");
        if (sys == 1)
            return new WinFactory();
        else
            return new LinuxFactory();
    }

    private static int readFromConfigFile(String key) {
        return 0;
    }

    public abstract Button createButton();
    public abstract Menu createMenu();
}
